File: go/twelve-days/twelve_days.go
package twelve

import (
	"fmt"
	"strings"
)

var verses = map[int][]string{
	1:  {"first", "and a Partridge in a Pear Tree."},
	2:  {"second", "two Turtle Doves"},
	3:  {"third", "three French Hens"},
	4:  {"fourth", "four Calling Birds"},
	5:  {"fifth", "five Gold Rings"},
	6:  {"sixth", "six Geese-a-Laying"},
	7:  {"seventh", "seven Swans-a-Swimming"},
	8:  {"eighth", "eight Maids-a-Milking"},
	9:  {"ninth", "nine Ladies Dancing"},
	10: {"tenth", "ten Lords-a-Leaping"},
	11: {"eleventh", "eleven Pipers Piping"},
	12: {"twelfth", "twelve Drummers Drumming"},
}

//Verse recites a particular line verse in the song "Twelve Days of Christmas"
func Verse(verseNumber int) string {
	suffix := []string{}
	for i := verseNumber; i > 0; i-- {
		suffix = append(suffix, verses[i][1])
	}
	if verseNumber == 1 {
		suffix = []string{"a Partridge in a Pear Tree."}
	}
	completeSuffix := strings.Join(suffix, ", ")
	return fmt.Sprintf("On the %s day of Christmas my true love gave to me: %s", verses[verseNumber][0], completeSuffix)
}

//Song recites the song "Twelve Days of Christmas" in it's entirity
func Song() string {
	song := []string{}
	for i := 1; i < 13; i++ {
		song = append(song, Verse(i))
	}
	return strings.Join(song, "\n") + "\n"
}

File: python/twelve-days/twelve_days.py
day_and_gift = [
    None,
    ["first", "and a Partridge in a Pear Tree."],
    ["second", "two Turtle Doves"],
    ["third", "three French Hens"],
    ["fourth", "four Calling Birds"],
    ["fifth", "five Gold Rings"],
    ["sixth", "six Geese-a-Laying"],
    ["seventh", "seven Swans-a-Swimming"],
    ["eighth", "eight Maids-a-Milking"],
    ["ninth", "nine Ladies Dancing"],
    ["tenth", "ten Lords-a-Leaping"],
    ["eleventh", "eleven Pipers Piping"],
    ["twelfth", "twelve Drummers Drumming"],
]


def verse(verse):
    if verse == 1:
        gifts = ["a Partridge in a Pear Tree."]
    else:
        v = verse
        gifts = []
        while v > 0:
            gifts.append(day_and_gift[v][1])
            v -= 1

    return f"On the {day_and_gift[verse][0]} day of Christmas my true love gave to me: {', '.join(gifts)}"


def recite(start_verse, end_verse):
    return [verse(i) for i in range(start_verse, end_verse + 1)]
